recursive-git-log-plugin/src/test/kotlin/com/linecorp/support/project/multi/log/git/recursive/FunctionalTest.kt

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.support.project.multi.log.git.recursive
+
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import org.gradle.testkit.runner.GradleRunner
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Test
+import java.io.File
+
+class FunctionalTest {
+
+    @Test
+    fun test() {
+        val here = File("").absolutePath
+
+        val path = File("$here/src/test/resources/sample")
+        val expectedFilePath = File("$here/src/test/resources/assertion")
+
+        GradleRunner.create()
+            .withProjectDir(path)
+            .withPluginClasspath()
+            .withArguments("clean")
+            .withGradleVersion("6.2")
+            .build()
+            .also { println(it.output) }
+
+        GradleRunner.create()
+            .withProjectDir(path)
+            .withPluginClasspath()
+            .withArguments("gitLog", "-Plog.git.from=v0.0.1", "-Plog.git.to=v0.0.2")
+            .withGradleVersion("6.2")
+            .build()
+            .also { println(it.output) }
+
+        GradleRunner.create()
+            .withProjectDir(path)
+            .withPluginClasspath()
+            .withArguments("gitLog", "-Plog.git.from=v0.0.2", "-Plog.git.to=v0.0.3")
+            .withGradleVersion("6.2")
+            .build()
+            .also { println(it.output) }
+
+        GradleRunner.create()
+            .withProjectDir(path)
+            .withPluginClasspath()
+            .withArguments("gitLog", "-Plog.git.from=v0.0.1", "-Plog.git.to=v0.0.3")
+            .withGradleVersion("6.2")
+            .build()
+            .also { println(it.output) }
+
+        File("$path/build/change_log_v0.0.1_v0.0.2.log").validate("$expectedFilePath/change_log_v0.0.1_v0.0.2.log")
+        File("$path/build/change_log_v0.0.2_v0.0.3.log").validate("$expectedFilePath/change_log_v0.0.2_v0.0.3.log")
+        File("$path/build/change_log_v0.0.1_v0.0.3.log").validate("$expectedFilePath/change_log_v0.0.1_v0.0.3.log")
```

Comment:
Parameterized tests can be used for these.

Reply:
It's hard to do it.
Because each tests shares build directory under resources.
If we wanna make this test parameterized, then we need to execute git log task in `@BeforeAll`.